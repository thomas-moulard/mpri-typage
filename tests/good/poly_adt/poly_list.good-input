type list 'a = 
   Nil 
|  Cons of 'a * list 'a
end

val nil = Nil

val l = Cons 0 nil
