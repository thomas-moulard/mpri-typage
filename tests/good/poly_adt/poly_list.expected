Nil = Nil

Cons = Cons

nil = Nil

l = Cons 0 Nil

l : list int
nil : forall 'a.list 'a
Cons : forall 'a.('a -> (list 'a -> list 'a))
Nil : forall 'a.list 'a
